builder/image/templates/builder

```diff
@@ -112,8 +106,10 @@ if [ ! -f Dockerfile ]; then
     BUILD_OPTS+=' -v '
     BUILD_OPTS+=$(echo $CACHE_DIR)
     BUILD_OPTS+=':/tmp/cache:rw'
-    # give non-root slugbuilder user R/W perms for docker volumes
-    chown -R 2000:2000 $TMP_DIR $CACHE_DIR
+    # give slug group ownership of TMP_DIR and CACHE_DIR. This is used by the
+    # slug builder. Contents in these folders will be owned by slug:slug.
```

Comment:
So, I checked this a bit further...

`TMP_DIR` is owned by `git:slug`, but the contents inside the directory is owned by `git:git`.
`CACHE_DIR` is owned by git:slug, and the contents (`.m2` / `.maven`) is owned by `slug:slug`.

I'll update the commit once some manual testing has been done, in case you're able to uncover any issues I havn't solved so far.